### PR TITLE
v-rmanda/183969/SampleSDK/AppOnly_Writes_SDK_sample

### PR DESCRIPTION
**Pull request by v-rmanda**
Made changes for ModuleWorkFlow.cs and program.cs for Apponly

### CODE REVIEW

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs [hunk 1]
```diff
@@ -25,12 +25,12 @@ public ModuleWorkflow(IConfiguration configuration)
         /// <summary>
         /// Workflow to create and publish the module
         /// </summary>
-        public async Task ClassworkAsync()
+        public async Task ClassworkAsync(bool appOnly = false)
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs]: add the parameter description in this header, similar to this sample:
https://github.com/microsoft/edu-assignments-graph-sdk/blob/7aca8b5a578476ed0373bb0f6133478fa5f7391c/samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs#L28C1-L29C127

Comment (cristobal-buenrostro) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs]: <param name="appOnly">True value authenticates the graph client with application permissions only, otherwise it will be created with delegated permissions</param>

Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs]: Added the below line

"/// <param name="appOnly">True value authenticates the graph client with application permissions only, otherwise it will be created with delegated permissions</param> "

Comment (sharad-sharma-msft) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs:29]: do we have similar options elsewhere in other code samples where app only is supported?

Comment (sharad-sharma-msft) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs:29]: somewhere in the readme, we should also document if the sample supports delegate or both delegate and app only

**Review by sharad-sharma-msft — CHANGES_REQUESTED** [review]

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs (hunk 1 above)
Comment (cristobal-buenrostro) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs]: please double check, the change is not reflected

Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs]: Push operation dint happen as expected, pushed the changes to PR.

Hence resolving

Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs:29]: Hi Sharad

This is the first time we are adding the appOnly for samples.

Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs:29]: > somewhere in the readme, we should also document if the sample supports delegate or both delegate and app only

Updated the ReadMe file with permission details

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs:1]: @v-rmanda 
I still don't see the README file change, please double check

Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs:1]: Pushed changed of ReadMe to PR.

**Review by cristobal-buenrostro — APPROVED** [review]

File: samples/csharp/MicrosoftEduGraphSamples/Program.cs [hunk 2]
```diff
@@ -14,5 +14,6 @@
 
 // Create an instance of the class you want to test and call the desired workflow method.
 // Each flow represents a common entire process that can be tested using Microsoft Graph SDK, all the sample flows are located in the "workflows" folder.
-ModuleWorkflow workflow = new ModuleWorkflow(config);
-await workflow.ClassworkAsync();
\ No newline at end of file
+
+ ModuleWorkflow workflow = new ModuleWorkflow(config);
+ await workflow.ClassworkAsync(true);
```
Comment (sharaaU7) [inline samples/csharp/MicrosoftEduGraphSamples/Program.cs]: appOnly: true

**Review by sharaaU7 — CHANGES_REQUESTED** [review]

File: samples/csharp/MicrosoftEduGraphSamples/Program.cs (hunk 2 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Program.cs]: Updated to "await workflow.ClassworkAsync(appOnly:true);"

**Review by sharaaU7 — APPROVED** [review]

File: README.md
Comment (cristobal-buenrostro) [inline README.md:1]: this PR was approved for the classwork sample. The changes you pushed today must be in a separate PR

File: samples/csharp/MicrosoftEduGraphSamples/Program.cs [hunk 3]
```diff
@@ -14,5 +14,7 @@
 
 // Create an instance of the class you want to test and call the desired workflow method.
 // Each flow represents a common entire process that can be tested using Microsoft Graph SDK, all the sample flows are located in the "workflows" folder.
-ModuleWorkflow workflow = new ModuleWorkflow(config);
-await workflow.ClassworkAsync();
\ No newline at end of file
+
+AssignmentWorkflow workflow = new AssignmentWorkflow(config);
+// await workflow.ClassworkAsync(appOnly:true);
+await workflow.CreateAndPatchAssignment(appOnly: false);
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Program.cs]: remove this line

File: samples/csharp/MicrosoftEduGraphSamples/Program.cs [hunk 4]
```diff
@@ -14,5 +14,7 @@
 
 // Create an instance of the class you want to test and call the desired workflow method.
 // Each flow represents a common entire process that can be tested using Microsoft Graph SDK, all the sample flows are located in the "workflows" folder.
-ModuleWorkflow workflow = new ModuleWorkflow(config);
-await workflow.ClassworkAsync();
\ No newline at end of file
+
+AssignmentWorkflow workflow = new AssignmentWorkflow(config);
+// await workflow.ClassworkAsync(appOnly:true);
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Program.cs]: uncomment this line

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 5]
```diff
@@ -34,7 +34,7 @@ public async Task<IEnumerable<EducationAssignment>> GetMeAssignmentsFromNonArchi
                 string userAccount = isTeacher ? _config["teacherAccount"] : _config["studentAccount"];
 
                 // Get a Graph client using delegated permissions
-                var graphClient = GraphClient.GetDelegateClient(_config["tenantId"], _config["appId"], userAccount, _config["password"]);
+                var graphClient = MicrosoftGraphSDK.GraphClient.GetDelegateClient(_config["tenantId"], _config["appId"], userAccount, _config["password"]);
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: remove the `MicrosoftGraphSDK.`

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 6]
```diff
@@ -123,5 +123,41 @@ public async Task AssignmentsEvolvableEnumsAsync()
                 Console.WriteLine($"AssignmentsEvolvableEnumsAsync: {ex.ToString()}");
             }
         }
-    }
+
+        /// <summary>
+        /// Workflow to get assignments from all the classes which are not archived
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: revise the description

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 7]
```diff
@@ -123,5 +123,41 @@ public async Task AssignmentsEvolvableEnumsAsync()
...
+        /// </summary>
+        /// <param name="isTeacher">True value accepts Teacher account and false for Student account</param>
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: revise the parameter name and the description

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 8]
```diff
@@ -123,5 +123,41 @@ public async Task AssignmentsEvolvableEnumsAsync()
...
+        /// Workflow to get assignments from all the classes which are not archived
+        /// </summary>
+        /// <param name="isTeacher">True value accepts Teacher account and false for Student account</param> 
+        public async Task CreateAndPatchAssignment(bool appOnly = false)
+        {
+            try
+            {
+                string assignmentId = string.Empty;
+
+                // Get a Graph client using delegated permissions
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: // Get a Graph client based on the appOnly parameter

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 9]
```diff
@@ -123,5 +123,41 @@ public async Task AssignmentsEvolvableEnumsAsync()
...
+                var graphClient = appOnly ? GraphClient.GetApplicationClient(_config["tenantId"], _config["appId"], _config["secret"]) : GraphClient.GetDelegateClient(_config["tenantId"], _config["appId"], _config["teacherAccount"], _config["password"]);
+
+                // Create assignment
+                var assignment = await Assignment.CreateAsync(graphClient, _config["classId"]);
+                assignmentId = assignment.Id;
+                Console.WriteLine($"Assignment created successfully {assignment.Id} in state {assignment.Status}");
+
+                //Creating a draft assignment
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: // Updating the draft assignment

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 10]
```diff
@@ -123,5 +123,41 @@ public async Task AssignmentsEvolvableEnumsAsync()
...
+                assignment = await Assignment.PatchAsync(graphClient, _config["classId"], assignmentId);
+                
+                //Verifying whether the DisplayName parameter is updated for the draft assignment.
+                assignment = await Assignment.GetAssignmentAsync(graphClient, _config["classId"], assignmentId);
+                if(assignment.DisplayName.Contains("updated"))
+                {
+                Console.WriteLine($"DisplayName updated successfully {assignment.Id} DisplayName {assignment.DisplayName}");
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: add tabulation

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs [hunk 11]
```diff
@@ -25,12 +25,13 @@ public ModuleWorkflow(IConfiguration configuration)
         /// <summary>
         /// Workflow to create and publish the module
         /// </summary>
-        public async Task ClassworkAsync()
+        /// <param name="appOnly">True value authenticates the graph client with application permissions only, otherwise it will be created with delegated permissions.</param> 
+        public async Task ClassworkAsync(bool appOnly = false)
         {
             try
             {
                 // Get a Graph client using delegated permissions
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs]: // Get a Graph client based on the appOnly parameter

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs [hunk 12]
```diff
@@ -121,6 +122,6 @@ public async Task ClassworkAsync()
                 Console.WriteLine($"ClassworkAsync: {ex.ToString()}");
                 return;
             }
-        }
+        }
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs]: nit: remove the trailing spaces

File: samples/csharp/MicrosoftGraphSDK/Assignment.cs [hunk 13]
```diff
@@ -239,5 +239,48 @@ public class Assignment
                 throw new GraphException($"DeactivateAsync call: {ex.Message}", ex, classId, assignmentId);
             }
         }
+
+        //Create a draft assignment
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftGraphSDK/Assignment.cs]: add a proper header as the other methods in this file

File: samples/csharp/MicrosoftGraphSDK/Assignment.cs [hunk 14]
```diff
@@ -239,5 +239,48 @@ public class Assignment
...
+        {
+            try
+            {
+              
+                var requestBody = new EducationAssignment
+                {
+                    DisplayName = "Reading and review test updated",                    
+                };
+
+                // Get a Graph client using delegated permissions
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftGraphSDK/Assignment.cs]: remove this comment

File: samples/csharp/MicrosoftGraphSDK/Assignment.cs [hunk 15]
```diff
@@ -239,5 +239,48 @@ public class Assignment
...
+                return await client.Education
+                    .Classes[classId]
+                    .Assignments[assignmentId]
+                    .PatchAsync(requestBody);
+
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftGraphSDK/Assignment.cs]: remove the blank line

File: samples/csharp/MicrosoftGraphSDK/Assignment.cs [hunk 16]
```diff
@@ -239,5 +239,48 @@ public class Assignment
...
+        //Create a draft assignment 
+        public static async Task<EducationAssignment> PatchAsync(
+            GraphServiceClient client,
+            string classId,
+            string assignmentId)
+        {
+            try
+            {
+
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftGraphSDK/Assignment.cs]: nit: remove the blank line

File: samples/csharp/MicrosoftGraphSDK/Assignment.cs [hunk 17]
```diff
@@ -239,5 +239,48 @@ public class Assignment
...
+                    .PatchAsync(requestBody);
+
+            }
+            catch (Exception ex)
+            {
+                throw new GraphException($"PatchAsync call: {ex.Message}", ex, classId, assignmentId);
+            }
+        }
+
+        //Delete the created assignment
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftGraphSDK/Assignment.cs]: add a proper header as the other methods in this file

**Review by cristobal-buenrostro — CHANGES_REQUESTED** [review]

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 18]
```diff
@@ -123,5 +123,41 @@ public async Task AssignmentsEvolvableEnumsAsync()
                 Console.WriteLine($"AssignmentsEvolvableEnumsAsync: {ex.ToString()}");
             }
         }
-    }
+
+        /// <summary>
+        /// Workflow to get assignments from all the classes which are not archived
+        /// </summary>
+        /// <param name="isTeacher">True value accepts Teacher account and false for Student account</param> 
+        public async Task CreateAndPatchAssignment(bool appOnly = false)
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: CreateAndPatchAssignmentAsync

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 19]
```diff
@@ -123,5 +123,41 @@ public async Task AssignmentsEvolvableEnumsAsync()
...
+                if(assignment.DisplayName.Contains("updated"))
+                {
+                Console.WriteLine($"DisplayName updated successfully {assignment.Id} DisplayName {assignment.DisplayName}");
+                }
+                await Assignment.DeleteAsync(graphClient, _config["classId"], assignmentId);
+
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"AssignmentsEvolvableEnumsAsync: {ex.ToString()}");
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: CreateAndPatchAssignmentAsync

File: samples/csharp/MicrosoftEduGraphSamples/Program.cs (hunk 3 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Program.cs]: removed the line "await workflow.CreateAndPatchAssignment(appOnly: false); "

File: samples/csharp/MicrosoftEduGraphSamples/Program.cs (hunk 4 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Program.cs]: uncommented the line "await workflow.ClassworkAsync(appOnly:true);

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 5 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: removed the "MicrosoftGraphSDK."

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 6 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: Updated the description to "Workflow to get create and Patch Assignment"

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 8 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: Changed to "// Get a Graph client using appOnly Parameter"

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 9 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: Changed to "//Updating a draft assignment"

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 10 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: added tabulation to if block

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs (hunk 11 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs]: updated the comment to "// Get a Graph client based on the appOnly parameter" on line 33

File: samples/csharp/MicrosoftGraphSDK/Assignment.cs (hunk 13 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftGraphSDK/Assignment.cs]: Updated the header to "/// <summary>
/// Create an update of Assignment
/// Reference :: https://learn.microsoft.com/en-us/graph/assignments-states-transition
/// </summary>
/// <param name="client">Microsoft Graph service client</param>
/// <param name="classId">User class id</param>
/// <param name="assignmentId">Assignment id in the class</param>
/// <returns>EducationAssignment</returns> " on line 243

File: samples/csharp/MicrosoftGraphSDK/Assignment.cs (hunk 14 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftGraphSDK/Assignment.cs]: removed the line 257 and a blank line space

File: samples/csharp/MicrosoftGraphSDK/Assignment.cs (hunk 15 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftGraphSDK/Assignment.cs]: removed blank line on line 262

File: samples/csharp/MicrosoftGraphSDK/Assignment.cs (hunk 16 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftGraphSDK/Assignment.cs]: removed blank line

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 18 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: updated the method name to "CreateAndPatchAssignmentAsync"

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 19 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: updated to "CreateAndPatchAssignmentAsync"

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs (hunk 12 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs]: removed trailing spaces

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 7 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: updated the parameter name and description for appOnly

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 20]
```diff
@@ -123,5 +123,43 @@ public async Task AssignmentsEvolvableEnumsAsync()
                 Console.WriteLine($"AssignmentsEvolvableEnumsAsync: {ex.ToString()}");
             }
         }
-    }
+
+        /// <summary>
+        /// Workflow to get create and Patch Assignment
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]:
```suggestion
        /// Workflow to create and patch Assignment
```

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 21]
```diff
@@ -123,5 +123,43 @@ public async Task AssignmentsEvolvableEnumsAsync()
...
+        /// </summary>
+        /// <param name="appOnly">True value authenticates the graph client with application permissions only, otherwise it will be created with delegated permissions.</param> 
+        public async Task CreateAndPatchAssignmentAsync(bool appOnly = false)
+        {
+            try
+            {
+                string assignmentId = string.Empty;
+
+                // Get a Graph client using appOnly Parameter
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]:
```suggestion
                // Get a Graph client based on the appOnly parameter
```

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 22]
```diff
@@ -123,5 +123,43 @@ public async Task AssignmentsEvolvableEnumsAsync()
...
+                //Updating a draft assignment
+                assignment = await Assignment.PatchAsync(graphClient, _config["classId"], assignmentId);
+                
+                //Verifying whether the DisplayName parameter is updated for the draft assignment.
+                assignment = await Assignment.GetAssignmentAsync(graphClient, _config["classId"], assignmentId);
+
+                    if(assignment.DisplayName.Contains("updated"))
+                    {
+                    Console.WriteLine($"DisplayName updated successfully {assignment.Id} DisplayName {assignment.DisplayName}");
+                    }
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]:
```suggestion
                if(assignment.DisplayName.Contains("updated"))
                {
                    Console.WriteLine($"DisplayName updated successfully {assignment.Id} DisplayName {assignment.DisplayName}");
                }
```

File: samples/csharp/MicrosoftEduGraphSamples/Program.cs [hunk 23]
```diff
@@ -14,5 +14,6 @@
 
 // Create an instance of the class you want to test and call the desired workflow method.
 // Each flow represents a common entire process that can be tested using Microsoft Graph SDK, all the sample flows are located in the "workflows" folder.
-ModuleWorkflow workflow = new ModuleWorkflow(config);
-await workflow.ClassworkAsync();
\ No newline at end of file
+
+AssignmentWorkflow workflow = new AssignmentWorkflow(config);
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Program.cs]:
```suggestion
ModuleWorkflow workflow = new ModuleWorkflow(config);
```

File: samples/csharp/MicrosoftGraphSDK/Assignment.cs [hunk 24]
```diff
@@ -239,5 +239,60 @@ public class Assignment
                 throw new GraphException($"DeactivateAsync call: {ex.Message}", ex, classId, assignmentId);
             }
         }
+
+        /// <summary>
+        /// Create an update of Assignment
+        /// Reference :: https://learn.microsoft.com/en-us/graph/assignments-states-transition
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftGraphSDK/Assignment.cs]:
```suggestion
        /// Update an assignment
```

File: samples/csharp/MicrosoftGraphSDK/Assignment.cs [hunk 25]
```diff
@@ -239,5 +239,60 @@ public class Assignment
...
+            }
+            catch (Exception ex)
+            {
+                throw new GraphException($"PatchAsync call: {ex.Message}", ex, classId, assignmentId);
+            }
+        }
+
+        /// <summary>
+        /// Delete the ncreated assignment
+        /// Reference :: https://learn.microsoft.com/en-us/graph/assignments-states-transition
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftGraphSDK/Assignment.cs]:
```suggestion
        /// Delete an assignment
```

File: samples/csharp/MicrosoftGraphSDK/Assignment.cs [hunk 26]
```diff
@@ -239,5 +239,60 @@ public class Assignment
...
+        /// </summary>
+        /// <param name="client">Microsoft Graph service client</param>
+        /// <param name="classId">User class id</param>
+        /// <param name="assignmentId">Assignment id in the class</param>
+        /// <returns>EducationAssignment</returns>
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftGraphSDK/Assignment.cs]:
```suggestion
        /// <returns></returns>
```

**Comment by cristobal-buenrostro** [comment]
@v-rmanda 
the rest of the comments are suggestions, please apply them.

**Review by cristobal-buenrostro — APPROVED** [review]

File: samples/csharp/MicrosoftGraphSDK/Assignment.cs [hunk 27]
```diff
@@ -239,5 +239,58 @@ public class Assignment
...
+        public static async Task<EducationAssignment> PatchAsync(
+            GraphServiceClient client,
+            string classId,
+            string assignmentId)
+        {
+            try
+            {              
+                var requestBody = new EducationAssignment
+                {
```
Comment (sharad-sharma-msft) [inline samples/csharp/MicrosoftGraphSDK/Assignment.cs]: pass in the requestBody

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 28]
```diff
@@ -123,5 +123,43 @@ public async Task AssignmentsEvolvableEnumsAsync()
                 Console.WriteLine($"AssignmentsEvolvableEnumsAsync: {ex.ToString()}");
             }
         }
-    }
+
+        /// <summary>
+        /// Workflow to create and patch Assignment
```
Comment (sharad-sharma-msft) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: Workflow to create a new assignment, followed by an update to the assignment display name (aka title)

File: samples/csharp/MicrosoftGraphSDK/Assignment.cs [hunk 29]
```diff
@@ -239,5 +239,58 @@ public class Assignment
...
+        public static async Task DeleteAsync(
+            GraphServiceClient client,
+            string classId,
+            string assignmentId)
+        {
+            try
+            {               
+                await client.Education.Classes[classId].Assignments[assignmentId].DeleteAsync();          
+            }
+            catch (Exception ex)
```
Comment (sharad-sharma-msft) [inline samples/csharp/MicrosoftGraphSDK/Assignment.cs:287]: what happens when there is no assignment with given id existing in the given class? shall we log 404 and not throw exception.

**Review by sharad-sharma-msft — CHANGES_REQUESTED** [review]

File: samples/csharp/MicrosoftGraphSDK/Assignment.cs (hunk 29 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftGraphSDK/Assignment.cs:287]: We are trying to delete the assignment created above due to that we might not see the above scenario. 
Probably we can cover error handling in a separate PR.

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 28 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: updated to "Workflow to create a new assignment, followed by an update to the assignment display name (aka title)"

File: samples/csharp/MicrosoftGraphSDK/Assignment.cs (hunk 27 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftGraphSDK/Assignment.cs]: passed in the requestbody

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 30]
```diff
@@ -123,5 +123,47 @@ public async Task AssignmentsEvolvableEnumsAsync()
...
+        {
+            try
+            {
+                string assignmentId = string.Empty;
+
+                // Get a Graph client based on the appOnly parameter
+                var graphClient = appOnly ? GraphClient.GetApplicationClient(_config["tenantId"], _config["appId"], _config["secret"]) : GraphClient.GetDelegateClient(_config["tenantId"], _config["appId"], _config["teacherAccount"], _config["password"]);
+
+                // Create assignment
+                var assignment = await Assignment.CreateAsync(graphClient, _config["classId"]);
```
Comment (sharad-sharma-msft) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: either call it CreateSampleAsync or pass in the request body here

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs [hunk 31]
```diff
@@ -123,5 +123,47 @@ public async Task AssignmentsEvolvableEnumsAsync()
                 Console.WriteLine($"AssignmentsEvolvableEnumsAsync: {ex.ToString()}");
             }
         }
-    }
+
+        /// <summary>
+        /// Workflow to create a new assignment, followed by an update to the assignment display name (aka title)
```
Comment (sharad-sharma-msft) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: Instead of workflow we should say "a code sample" across all SDK samples

**Review by sharad-sharma-msft — CHANGES_REQUESTED** [review]

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 31 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: updated "workflow" to "A code sample" across all SDK Samples

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs (hunk 30 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs]: replaced "CreateAsync" to "CreateSampleAsync"

**Review by sharad-sharma-msft — APPROVED** [review]
